xt_quirk: link devel sources instead copy

varder opened the pull request:
Use symbol link for development sources instead of copy

PR should be merged with https://github.com/xen-troops/meta-xt-prod-devel/pull/327

@andr2000
@lorc

Signed-off-by: Valerii Chubar <valerii_chubar@epam.com>

andr2000 commented on classes/xt_quirks.bbclass:
```diff
@@ -44,18 +41,14 @@ python do_unpack_xt_extras() {
+        try:
+            type, _, location, _, _, _ = bb.fetch.decodeurl(url)
+            path = bb.fetch2.localpath(url, d)
+            bb.debug(1, "Make symbol link for path:" + path)
```
s/symbol/symbolic

andr2000 reviewed (approved):
After proper testing and nit fixed:
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`

Commit:
xt_quirks: force compile and configure

After the domain image is built, an attempt to run the do_compile
or do_configure command, for example:
'bitbake domd-image-weston -c compile'
will not have any effect.

Ignore the timestamps for the 'compile' and 'configure'.
Consequently, "nostamp" tasks are always rerun, updating
domain configuration files and building images.

Signed-off-by: Valerii Chubar <valerii_chubar@epam.com>
Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>

Commit:
xt_quirk: link devel sources instead copy

XT_QUIRK_UNPACK_SRC_URI is used to copy additional files into the inner
(domain's) build. For development purposes for every change in the
repository, it is necessary to move files into internal assemblies.

Use symbol links instead of copying to use recipe source code directly
from the original repository.

Signed-off-by: Valerii Chubar <valerii_chubar@epam.com>
Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>

andr2000 commented on classes/xt_quirks.bbclass:
```diff
@@ -100,3 +100,6 @@ python do_xt_config_reconstruct() {
         f.write("require" + " " + "build-versions.inc" + "\n")
 }
+
+do_configure[nostamp] = "1"
```
Will do_configure remove the lower Yocto and start the build from the start (with state caches used?)

varder replied:
I have tested and in this case do_configure does not cause lower yocto removal.

andr2000 replied:
looks good then

andr2000 commented on classes/xt_quirks.bbclass:
```diff
@@ -44,18 +41,24 @@ python do_unpack_xt_extras() {
 
     for url in urls:
-        type, _, location, _, _, _ = bb.fetch.decodeurl(url)
-        item = check_url_or_pack(url, type, location, d) + " "
```
check_url_or_pack is no longer needed

varder replied:
done

andr2000 commented on classes/xt_quirks.bbclass:
```diff
@@ -44,18 +41,24 @@ python do_unpack_xt_extras() {
-    urls = (d.getVar("XT_QUIRK_UNPACK_SRC_URI") or "").split()
-    for url in urls:
-        type, _, location, _, _, _ = bb.fetch.decodeurl(url)
-        remove_tarfile(url, type, location, d)
```
remove_tarfile is no longer needed

varder replied:
done

rshym commented:
Tested by build of prod-devel (with https://github.com/xen-troops/meta-xt-prod-devel/pull/327) and prod-aos (with https://github.com/xen-troops/meta-xt-prod-aos/pull/111).
Products built successfully.
